yt/geometry/geometry_handler.py

```diff
@@ -464,6 +465,15 @@ def __next__(self):
 
 def is_curvilinear(geo):
     # tell geometry is curvilinear or not
+    issue_deprecation_warning(
+        "the is_curvilear() function is deprecated. "
+        "Instead, compare the geometry object directly with yt.geometry.geometry_enum.Geometry "
+        "enum members, as for instance:\n"
+        "if is_curvilinear(geometry):\n    ...\n"
+        "should be rewritten as:"
+        "if geometry is Geometry.POLAR or geometry is Geometry.CYLINDRICAL or geometry is Geometry.SPHERICAL:\n    ...",
```

Comment:
because it was written at a time we had only "cartesian", "polar", "cylindrical" and "spherical" geometries, but not "geographic", which IMO could and probably should be considered curvilinear, but the function was never updated, so I think deprecating it is a better option than taking a risk to break user code that may rely on the current behaviour.